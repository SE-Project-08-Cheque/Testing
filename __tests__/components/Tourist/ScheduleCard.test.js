import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import ScheduleCard from "../../../components/Tourist/ScheduleCard";

afterEach(cleanup);

it("renders schedule card correctly", () => {
  const { getByTestId } = render(
    <ScheduleCard
      place="Hikkaduwa"
      date="24-02-2021"
      travellingMethod="bus"
      fair="100"
      state="true"
    />
  );
  expect(getByTestId("place")).toHaveTextContent("Hikkaduwa");
});

it("schedule card matches snapshot", () => {
  const tree = renderer.create(
    <ScheduleCard
      place="Hikkaduwa"
      date="24-02-2021"
      travellingMethod="bus"
      fair="100"
      state="true"
    />
  );

  expect(tree).toMatchSnapshot();
});
