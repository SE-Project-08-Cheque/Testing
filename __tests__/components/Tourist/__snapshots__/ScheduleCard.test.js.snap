// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schedule card matches snapshot 1`] = `
<div
  className="css-mrjdax"
>
  <div
    className="css-n16x53"
  >
    <span
      className="chakra-badge css-637omx"
    >
      true
    </span>
    <h4
      className="css-1mwm0bw"
      data-testid="place"
    >
      Hikkaduwa
    </h4>
    <div
      className="css-0"
    >
      <p
        className="chakra-text css-0"
      >
        On : 24-02-2021
      </p>
      <p
        className="chakra-text css-0"
      >
        At : undefined
      </p>
      <p
        className="chakra-text css-0"
      >
        TravellingMethod : bus
      </p>
      <p
        className="chakra-text css-0"
      >
        Fair : 100
      </p>
    </div>
  </div>
</div>
`;
